flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcSystem.java

```diff
@@ -74,8 +87,39 @@ RpcServiceBuilder withExecutorConfiguration(
      * @return loaded RpcSystem
      */
     static RpcSystem load() {
-        final ClassLoader classLoader = RpcSystem.class.getClassLoader();
-        return ServiceLoader.load(RpcSystem.class, classLoader).iterator().next();
+        return load(new Configuration());
+    }
+
+    /**
+     * Loads the RpcSystem.
+     *
+     * @return loaded RpcSystem
+     */
+    static RpcSystem load(Configuration config) {
+        try {
+            final ClassLoader classLoader = RpcSystem.class.getClassLoader();
+
+            final String tmpDirectory = ConfigurationUtils.parseTempDirectories(config)[0];
+            final Path tempFile =
+                    Files.createFile(
+                            Paths.get(tmpDirectory, UUID.randomUUID() + "_flink-rpc-akka.jar"));
```

Comment:
It is a bit annoying that something from flink-rpc-core is now so hard-wired to flink-rpc-akka, so maybe we should the loading to flink-runtime.

Reply:
From intuition, I'd agree but I have no idea on the effort.

Reply:
No effort at all, and it would force all users of this method to depend on flink-runtime (without which it would fail anyway).